### PR TITLE
Adding README.

### PR DESCRIPTION
**Pull request by PiotrProkop**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,4 +5,76 @@
 [![Go Report Card](https://goreportcard.com/badge/stackanetes/kubernetes-entrypoint "Go Report Card")](https://goreportcard.com/report/stackanetes/kubernetes-entrypoint)
 
 
+============
+
+Kubernetes-entrypoint enables complex deployments on top of Kubernetes.
+
+## Overview
+
+Kubernetes-entrypoint needs to be deployed as a container entrypoint which means it has to bundled in container. Before launching desired application
```
Comment (Quentin-M) [inline README.md]: `Kubernetes-entrypoint is meant to be used as a container entrypoint, which means it has to bundled in the container. Before launching the desired application, the entrypoint verifies and waits for all specified dependencies to be met.`

File: README.md [hunk 2]
```diff
@@ -5,4 +5,76 @@
...
+it checks if all specified dependencies are met.
+
+Kubernetes-entrypoint talks directly to Kubernetes API so there is no need for centralized orchestration layer to manage deployments.
+Each container is self aware of its dependencies so thing like failure recovery or pods migration becomes really easy.
```
Comment (Quentin-M) [inline README.md]: > The Kubernetes-entrypoint queries directly the Kubernetes API and each container is self-aware of its dependencies and their states. Therefore, no centralized orchestration layer is required to manage deployments and scenarios such as failure recovery or pod migration become easy.

File: README.md [hunk 3]
```diff
@@ -5,4 +5,76 @@
...
+
+## Usage
+
+Kubernetes-entrypoint uses environment variables passed into a container.
```
Comment (Quentin-M) [inline README.md]: > Kubernetes-entrypoint reads the dependencies out of environment variables passed into a container.

File: README.md [hunk 4]
```diff
@@ -5,4 +5,76 @@
...
+There is only one required environment variable "COMMAND" which specifies a command(arguments delimited by whitespace) which has to be executed when all dependencies are resolved:
+
+```COMMAND="sleep inf"```
+
+Kubernetes-entrypoint introduces a range variety of dependencies which can be used to better orchestrate once deployment.
```
Comment (Quentin-M) [inline README.md]: s/range/wide/

File: README.md [hunk 5]
```diff
@@ -5,4 +5,76 @@
...
+
+## Dependencies
```
Comment (Quentin-M) [inline README.md]: Supported types of dependencies?

File: README.md [hunk 6]
```diff
@@ -5,4 +5,76 @@
...
+
+All dependencies are passed as environement variables in format of `DEPENDENCY_<NAME>` delimited by colon. For dependencies to be effective
+please use [readiness probes](http://kubernetes.io/docs/user-guide/production-pods/#liveness-and-readiness-probes-aka-health-checks) for all containers.
+
+- Service:
```
Comment (Quentin-M) [inline README.md]: Maybe use `####` to make them appear more clearly?

File: README.md [hunk 7]
```diff
@@ -5,4 +5,76 @@
...
+    checks whether given kubernetes service has at least one endpoint.  
+    Example:
+
+    ```DEPENDENCY_SERVICE=mariadb,keystone-api```
+
+- Container:
+    for multicontainer pod we can specify to wait with launching application on other container in the same pod,
+    this dependency requires a `POD_NAME` env which can be easily passed as [downward api](http://kubernetes.io/docs/user-guide/downward-api/).
```
Comment (Quentin-M) [inline README.md]: s/env/environment variable/

> passed through the `[downward api](http://kubernetes.io/docs/user-guide/downward-api/)`

File: README.md [hunk 8]
```diff
@@ -5,4 +5,76 @@
...
+
+- Config:
+    this dependency performs a container level templating of configuration files. It can template an ip address `{{ .IP }}` and hostname `{{ .HOSTNAME }}`. 
+    Templated config has to be stored in an arbitrary directory `/configmaps/<name_of_file>/<name_of_file>`.
+    This dependency requires `INTERFACE_NAME` environment variable to know which interface to use for obtain ip address. 
+    Example:
+
+    ```DEPENDENCY_CONFIG=/etc/nova/nova.conf```
+
+    kubernetes-entrypoint will look for config in /configmaps/nova.conf/nova.conf template `{{ .IP }} and {{ .HOSTNAME }}` and place it as
```
Comment (Quentin-M) [inline README.md]: > The Kubernetes-entrypoint will look for the configuration file `/configmaps/nova.conf/nova.conf`, template any `{{ .IP }} and {{ .HOSTNAME }}` tags and save the file as `/etc/nova/nova.conf`.

File: README.md [hunk 9]
```diff
@@ -5,4 +5,76 @@
...
+Kubernetes-entrypoint is meant to be used as a container entrypoint, which means it has to bundled in the container. 
+Before launching the desired application, the entrypoint verifies and waits for all specified dependencies to be met.
+
+The Kubernetes-entrypoint queries directly the Kubernetes API and each container is self-aware of its dependencies and their states. 
+Therefore, no centralized orchestration layer is required to manage deployments and scenarios such as failure recovery or pod migration become easy.
+
+## Usage
+
+Kubernetes-entrypoint reads the dependencies out of environment variables passed into a container.
+There is only one required environment variable "COMMAND" which specifies a command(arguments delimited by whitespace) which has to be executed when all dependencies are resolved:
```
Comment (DTadrzak) [inline README.md]: s/command(/command (/

File: README.md [hunk 10]
```diff
@@ -5,4 +5,76 @@
...
+
+`COMMAND="sleep inf"`
+
+Kubernetes-entrypoint introduces a wide variety of dependencies which can be used to better orchestrate once deployment.
+
+## Supported types of dependencies
+
+All dependencies are passed as environement variables in format of `DEPENDENCY_<NAME>` delimited by colon. For dependencies to be effective
+please use [readiness probes](http://kubernetes.io/docs/user-guide/production-pods/#liveness-and-readiness-probes-aka-health-checks) for all containers.
```
Comment (DTadrzak) [inline README.md]: unnecessary new line?

File: README.md [hunk 11]
```diff
@@ -5,4 +5,76 @@
...
+This dependency requires `INTERFACE_NAME` environment variable to know which interface to use for obtain ip address. 
+Example:
+
+`DEPENDENCY_CONFIG=/etc/nova/nova.conf`
+
+The Kubernetes-entrypoint will look for the configuration file `/configmaps/nova.conf/nova.conf`, template 
+`{{ .IP }} and {{ .HOSTNAME }}` tags and save the file as `/etc/nova/nova.conf`.
+
+### Socket
+Checks wheter a given file exists and container has rights to read it.
```
Comment (DTadrzak) [inline README.md]: s/wheter/whether/

**Pull request merged**